### ISSUE TITLE
LGBMRegressor doesn't learn with custom objective and metric

### ISSUE THREAD

**Issue by david26694** [microsoft/LightGBM#3584]
I'm trying to use a custom metric and objective for lightgbm. I'm trying to replicate the mean absolute error but the error doesn't seem to change. The custom metric and objective for MAE are defined in the example.

## How you are using LightGBM?

LightGBM component: Python package.

## Environment info

Operating System: macOS Catalina Version 10.15.2 (19C57)

CPU/GPU model: 1,4 GHz Quad-Core Intel Core i5

C++ compiler version: Apple clang version 11.0.3 (clang-1103.0.32.59)

CMake version:

Java version:

Python version: 3.7

R version:

Other:

LightGBM version or commit hash:

## Error message and / or logs

```
[10]    training's mae: 22.5328
[20]    training's mae: 22.5328
[30]    training's mae: 22.5328
[40]    training's mae: 22.5328
[50]    training's mae: 22.5328
[60]    training's mae: 22.5328
[70]    training's mae: 22.5328
[80]    training's mae: 22.5328
[90]    training's mae: 22.5328
[100]   training's mae: 22.5328
```

## Reproducible example(s)

```
import numpy as np
from lightgbm import LGBMRegressor
from sklearn.datasets import load_boston


def mae_objective(y, preds):

    grad = np.sign(preds - y)
    hess = np.zeros(len(y))

    return grad, hess


def mae_metric(y, preds):
    higher_is_better = False
    return 'mae', np.abs(preds - y).mean(), higher_is_better


if __name__ == "__main__":

    X, y = load_boston(return_X_y=True)
    # Using this, it learns
    # lgb = LGBMRegressor(objective='regression_l1')
    lgb = LGBMRegressor(objective=mae_objective, n_estimators=100)

    lgb.fit(
        X,
        y,
        eval_metric=mae_metric,
        eval_set=[(X, y),],
        verbose=10
    )

```

**Comment by StrikerRUS** [comment]
@david26694 Please refer to https://github.com/microsoft/LightGBM/issues/2128#issuecomment-492053052 and https://stackoverflow.com/a/58548572.
